### PR TITLE
Ofmcc 1626 supplementary review

### PR DESCRIPTION
**Pull request by jenbeckett**

### CODE REVIEW

File: frontend/src/components/supp-allowances/SupportNeedsSummary.vue [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+<script>
+import AppMissingInfoError from '@/components/ui/AppMissingInfoError.vue'
+import { APPLICATION_ERROR_MESSAGES } from '@/utils/constants'
+import { SUPPORT_CHECKBOX_LABELS } from './suppConstants'
+
+export default {
+  components: { AppMissingInfoError },
+  props: {
+    supportModel: {
+      type: Object,
```
Comment (jgstorey) [inline frontend/src/components/supp-allowances/SupportNeedsSummary.vue:34]: consider if supportModel should be required: true

File: frontend/src/components/supp-allowances/TransportationSummary.vue [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import AppMissingInfoError from '@/components/ui/AppMissingInfoError.vue'
+import AppLabel from '@/components/ui/AppLabel.vue'
+import { APPLICATION_ERROR_MESSAGES } from '@/utils/constants'
+
+export default {
+  components: { AppMissingInfoError, AppLabel },
+  props: {
+    transportModels: {
+      type: Array,
+      default: () => {
```
Comment (jgstorey) [inline frontend/src/components/supp-allowances/TransportationSummary.vue:101]: consider if prop should be required?

File: frontend/src/views/supp-allowances/SupplementarySubmitView.vue [hunk 3]
```diff
@@ -1,3 +1,308 @@
...
+    async loadData() {
+      try {
+        this.loading = true
+        this.$emit('process', true)
+        //this page should specifiy to load only those applications in "draft" status - as there will be
+        //scenarios where some applications have been submitted, but the user will want to come back and fill in others.
+        this.models = await ApplicationService.getSupplementaryApplications(this.$route.params.applicationGuid)
+
+        for (const el of this.models) {
+          const found = this.PANELS.find((panel) => panel.supplementaryType == el.supplementaryType)
```
Comment (jgstorey) [inline frontend/src/views/supp-allowances/SupplementarySubmitView.vue:226]: Optional: could refactor some areas of concern out of loadData for readable/maintainable... i.e. loadDocuments(model) and setPanelId(model)

**Review by jgstorey — CHANGES_REQUESTED** [review]
Small changes, some fyi/optional.

File: frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+<script>
+import AppMissingInfoError from '@/components/ui/AppMissingInfoError.vue'
+import { APPLICATION_ERROR_MESSAGES } from '@/utils/constants'
+import { INDIG_CHECKBOX_LABELS } from './suppConstants'
+
+export default {
+  components: { AppMissingInfoError },
+  props: {
+    indigenousProgrammingModel: {
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue:33]: done

File: frontend/src/components/supp-allowances/SupportNeedsSummary.vue (hunk 1 above)
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/SupportNeedsSummary.vue:34]: done

File: frontend/src/components/supp-allowances/TransportationSummary.vue (hunk 2 above)
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/TransportationSummary.vue:101]: it should be :) done

File: frontend/src/utils/constants.js [hunk 5]
```diff
@@ -93,6 +99,8 @@ export const APPLICATION_ERROR_MESSAGES = Object.freeze({
   FACILITY_TYPE: 'Facility type required',
   OPERATIONAL_COST: 'Operating costs required -or- Facility costs required',
   STAFFING: 'Staffing information required',
+  TRANSPORT: 'Transportation information required',
+  OTHER: 'Description of other costs required',
```
Comment (vietle-cgi) [inline frontend/src/utils/constants.js:103]: could we rename these constants to SUPP_TRANSPORT and SUPP_OTHER? so we will know these constants are for SUPP app

File: frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue [hunk 6]
```diff
@@ -67,25 +67,33 @@
...
+            </p>
+          </template>
+        </v-checkbox>
+      </v-col>
+    </v-row>
+  </div>
+  <div v-else>
+    <v-row class="ml-2 my-5">
+      <v-icon size="x-large" color="warning">mdi mdi-alert</v-icon>
+      <h4 class="ml-3">You must have an inclusion policy to apply for Support Needs Funding. Your organization account manager can update inclusion policy details in Account Management.</h4>
```
Comment (vietle-cgi) [inline frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue:94]: Should Account Management be a hyperlink to the Organization Management page?

**Review by jgstorey — APPROVED** [review]
Reviewed approved.

**Review by jgstorey — APPROVED** [review]

File: frontend/src/views/supp-allowances/SupplementaryConfirmation.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+<script>
+import AppButton from '@/components/ui/AppButton.vue'
+import AppBackButton from '@/components/ui/AppBackButton.vue'
+
+export default {
+  name: 'ApplicationConfirmationView',
+  components: { AppButton, AppBackButton },
+  computed: {},
+  created() {},
```
Comment (vietle-cgi) [inline frontend/src/views/supp-allowances/SupplementaryConfirmation.vue:23]: can we remove computed and created if they are unused?

**Review by vietle-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/utils/constants.js (hunk 5 above)
Comment (jenbeckett) [reply frontend/src/utils/constants.js:103]: sure :)

File: frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue (hunk 6 above)
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue:94]: Probably? The requirements said to link to "account management" - but maybe they mean Org Management?

File: frontend/src/views/supp-allowances/SupplementaryConfirmation.vue (hunk 7 above)
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryConfirmation.vue:23]: done

File: frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue (hunk 6 above)
Comment (vietle-cgi) [reply frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue:94]: Account Management is a generic name for both Org/Facility Management and User Managment. I see the hyperlink in wireframe so I guess we should have that? It should go to this page: "http://localhost:8082/account-mgmt/manage-organization"

Comment (vietle-cgi) [reply frontend/src/components/supp-allowances/SupportNeedsProgrammingAllowance.vue:94]: FYI, we have a similar hyperlink in ServiceDeliveryView

**Review by vietle-cgi — APPROVED** [review]

**Pull request merged**